google-cloud-clients/google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -254,17 +234,43 @@ public void onCompleted() {
                 "numDocuments", AttributeValue.longAttributeValue(documentReferences.length)));
 
     streamRequest(request.build(), responseObserver, firestoreClient.batchGetDocumentsCallable());
-
-    return futureList;
   }
 
-  @Nonnull
-  @Override
-  public void getAll(
-      @Nonnull DocumentReference[] documentReferences,
+  /** Internal getAll() method that accepts an optional transaction id. */
+  ApiFuture<List<DocumentSnapshot>> getAll(
+      final DocumentReference[] documentReferences,
       @Nullable FieldMask fieldMask,
-      final ApiStreamObserver<DocumentSnapshot> responseObserver) {
-    this.getAll(documentReferences, fieldMask, null, responseObserver);
+      @Nullable ByteString transactionId) {
+    final SettableApiFuture<List<DocumentSnapshot>> futureList = SettableApiFuture.create();
+    final Map<DocumentReference, DocumentSnapshot> documentSnapshotMap = new HashMap<>();
```

Comment:
HashMap would not maintain your insertion order... the default behavior is to have Instersion order right?

Reply:
logic is changed now no need to maintain insertion order